lib/Transforms/LayoutPropagation/LayoutPropagation.td

```diff
@@ -0,0 +1,114 @@
+#ifndef LIB_TRANSFORMS_LAYOUTPROPAGATION_LAYOUTPROPAGATION_TD_
```

Comment:
Could you place this transform under lib/Dialect/TensorExt/Transforms? If a transform is pretty dialect specific I think we'd like to put it there, to avoid making lib/Transforms/ such a mixed bag (@AlexanderViand-Intel and I still had leftover un-categorized transforms there, but this one seems pretty clearly under TensorExt)

Reply:
I actually had it there and decided, because of the dependence on ops from `tensor`, `tensor_ext` and `linalg`, that it should be considered a general transform not specific to one dialect. There will definitely be more `linalg` ops coming (convolution, matmul, matvec) and I could also see it supporting layout assignment for higher-level ML ops if they are present.